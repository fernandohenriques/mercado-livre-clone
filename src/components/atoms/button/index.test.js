import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from './index';

describe('<Button />', () => {
  const onClick = jest.fn();
  const buttonText = 'Comprar';

  it('Snapshot testing', () => {
    const { asFragment } = render(
      <Button onClick={onClick}>
        {buttonText}
      </Button>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('Should call onClick function if click in button', () => {
    const { getByTitle } = render(
      <Button onClick={onClick}>
        {buttonText}
      </Button>
    );

    const button = getByTitle(buttonText);
    fireEvent.click(button);

    expect(onClick).toBeCalled();
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
