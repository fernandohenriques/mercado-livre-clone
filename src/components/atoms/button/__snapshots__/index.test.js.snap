// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Snapshot testing 1`] = `
<DocumentFragment>
  <button
    class="button"
    title="Comprar"
    type="button"
  >
    Comprar
  </button>
</DocumentFragment>
`;
